### PR TITLE
Update React18 Fluent9 functional components and other fixes

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
**Change Description**:

1. Use React ver. 18 
2. Remove all react-hot-loader stuff because it is now built into React 18. 
3. Use functional components instead of class components and use React effects instead of lifecycle events for state management.
4. Use Fluent React UI v9.
5. Use the Fluent V9 "makeStyles" API for styling. 
6. Removed the unused "Commands" files and manifest markup.
7. Remove the testing of whether Office is initialized. It served no purpose.
8. Change the functionality of the resulting projects so that they show real world React state management. They all now insert text into the document.
9. Add check in taskpane.html for Trident (IE) and Edge Legacy and gracefully fail if positive. **I WOULD EXPECT THE IE AND EDGE LEGACY TESTS TO FAIL FOR THIS REASON. WE SHOULD PROBABLY JUST NOT HAVE THOSE TESTS FOR THIS TEMPLATE.**
10. Enforce _Separation of Concerns_ by moving code that calls Office.js out of the React component modules and into their own modules. 
11. Update all unit and end2end tests to account for the above. **THIS APPLIES ONLY TO THE WINDOWS TESTS. IF MAC AND WEB TESTS ALSO NEED TO CHANGE, I WILL NEED HELP WITH THAT.**
12. Use the actual production *.tsx component files for the end2end tests instead of a separate set of (near, but not exact) duplicate files.
13. Update convertToSingleHost.js to account for all of the above.
14. Remove code and files that would only be relevant if we were including tests in the emitted projects.

1. **Do these changes impact any *npm scripts* commands (in package.json)?** (e.g., running 'npm run start')
    If Yes, briefly describe what is impacted.
    No

2. **Do these changes impact *VS Code debugging* options (launch.json)?**
    If Yes, briefly describe what is impacted.
    No

3. **Do these changes impact *template output*?** (e.g., add/remove file, update file location, update file contents)
    If Yes, briefly describe what is impacted.
    Yes. See **Change Description** above.

4. **Do these changes impact *documentation*?** (e.g., a tutorial on https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins)
    If Yes, briefly describe what is impacted.
    Yes. The quickstart/tutorial for a React-based add-in will have to be changed and possibly other files too. **THIS PR SHOULD NOT BE MERGED TO PRODUCTION UNTIL THE DOC REVISIONS ARE READY TO BE MERGED TOO. NEEDS TO HAPPEN AT THE SAME TIME.**

If you answered yes to any of these please do the following:
    > Include 'Rick-Kirkham' in the review
    > Make sure the README file is correct

**Validation/testing performed**:

All unit and end2end tests work locally.
All emitted projects tested and they all work.

### CODE REVIEW

**Comment by akrantz** [comment]
@Rick-Kirkham It's better to create a different branch rather than use the master branch in your fork. You don't want to keep the extra commits in your fork and also merge back from the main repository especially if you use squash commit to complete the PR.

File: src/taskpane/components/TextInsertion.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { useState } from "react";
+import { Button, Field, Textarea, tokens, makeStyles } from "@fluentui/react-components";
+import { selectInsertionByHost } from "../../../test/end-to-end/src/host-relative-text-insertion";
```
Comment (akrantz) [inline src/taskpane/components/TextInsertion.tsx:4]: You cannot reference test code here. The test code can reference the add-in source code but not the other way around.

File: src/commands/commands.ts
Comment (akrantz) [inline src/commands/commands.ts:1]: Commands aren't just for Outlook. Ribbon buttons across all hosts can invoke button handlers. Wouldn't it be better to do the refactor for React functional components and Fluent and leave the support for commands in?

**Comment by akrantz** [comment]
Sorry I commented on the wrong PR accidentally.

File: src/taskpane/components/TextInsertion.tsx (hunk 1 above)
Comment (Rick-Kirkham) [reply src/taskpane/components/TextInsertion.tsx:4]: The update convertToSingleHost.js replaces this line, so it's not in the source code of the emitted projects.

File: src/commands/commands.ts
Comment (Rick-Kirkham) [reply src/commands/commands.ts:1]: Darren expressed the same idea. I will raise this question with my team and the DevX PM, Ben Siler, in October. My feeling is that Commands are just one of many things that developers could do to enhance the task pane add-in. There's no particular reason to favor them over the many other kinds of enhancements and create one in the project. I'd like to keep the projects that are initially created simple.

**Pull request closed without merging**